larray/core/array.py

```diff
@@ -5572,6 +5542,201 @@ def extend(self, axis, other):
+            Label(s) or group after which to insert `new_labels`.
+        pos : int or tuple/list of int.
+            Index/indices before which to insert `new_labels`.
+        axis : axis reference, optional
+            Axis in which to insert input array (`value`). This is only required when using `pos` or when after or
+            before are ambiguous labels.
+        label : str, optional
+            Label for the new item in axis.
+
+        Returns
+        -------
+        LArray
+            Array with `value` inserted along `axis`. The dtype of the returned array will be the "closest" type
+            which can hold both the array values and the inserted values without loss of information. For example,
+            when mixing numeric and string types, the dtype will be object.
+
+        Examples
+        --------
+        >>> arr1 = ndtest((2, 3))
+        >>> arr1
+        a\\b  b0  b1  b2
+         a0   0   1   2
+         a1   3   4   5
+        >>> arr1.insert(42, before='b1', label='b0.5')
+        a\\b  b0  b0.5  b1  b2
+         a0   0    42   1   2
+         a1   3    42   4   5
+        >>> arr2 = ndtest(2)
+        >>> arr2
+        a  a0  a1
+            0   1
+        >>> arr1.insert(arr2, after='b0', label='b0.5')
+        a\\b  b0  b0.5  b1  b2
+         a0   0     0   1   2
+         a1   3     1   4   5
+        >>> arr1.insert(42, axis='b', pos=1, label='b0.5')
+        a\\b  b0  b0.5  b1  b2
+         a0   0    42   1   2
+         a1   3    42   4   5
+        >>> arr1.insert(42, before=X.b.i[1], label='b0.5')
+        a\\b  b0  b0.5  b1  b2
+         a0   0    42   1   2
+         a1   3    42   4   5
+
+        insert an array which already has the axis
+
+        >>> arr3 = ndrange('a=a0,a1;b=b0.1,b0.2') + 42
+        >>> arr3
+        a\\b  b0.1  b0.2
+         a0    42    43
+         a1    44    45
+        >>> arr1.insert(arr3, before='b1')
+        a\\b  b0  b0.1  b0.2  b1  b2
+         a0   0    42    43   1   2
+         a1   3    44    45   4   5
+        """
+
+        # XXX: unsure we should have arr1.insert(arr3, before='b1,b2') result in (see unit tests):
+
+        # a\\b  b0  b0.1  b1  b0.2  b2
+        #  a0   0    42   1    43   2
+        #  a1   3    44   4    45   5
+
+        # we might to implement the following instead:
+
+        # a\\b  b0  b0.1  b0.2  b1  b0.1  b0.2  b2
+        #  a0   0    42    43   1    42    43   2
+        #  a1   3    44    45   4    44    45   5
+
+        # The later looks less useful and could be emulated easily via:
+        # arr1.insert([arr3, arr3], before='b1,b2')
+        # while the above is a bit harder to achieve manually:
+        # arr1.insert([arr3[[b]] for b in arr3.b], before=['b1', 'b2'])
+        # but the later is *probably* more intuitive (and wouldn't suffer from the inefficiency we currently have).
+
+        # XXX: when we have several lists, we implicitly match them by position, which we should avoid for the usual
```

Comment:
several values ? I'm not sure what you are talking about with "lists".

Reply:
when we pass a list (instead of a scalar) for several of "value", "before/after/pos" and "label".